test/integration/parachain/staging/sequential/redeem.test.ts

```diff
@@ -116,6 +118,44 @@ describe("redeem", () => {
         );
     });
 
+    it("should have applied oracle fee rate to redeem transaction", async () => {
+        const oracleBtcFeePerByte = await interBtcAPI.oracle.getBitcoinFees();
+
+        const redeemRequests = await interBtcAPI.redeem.list();
+        for (const redeemRequest of redeemRequests) {
+            const opreturnData = stripHexPrefix(redeemRequest.id.toString());
+            const txId = await interBtcAPI.electrsAPI.getTxIdByOpReturn(opreturnData)
+                .catch((e) => {
+                    // if no opreturn is ready, we may still have to wait for it
+                    return interBtcAPI.electrsAPI.waitForOpreturn(opreturnData, 5 * 60 * 1000, 5000);
```

Comment:
Since this returns immediately if a tx with the given OP_RETURN is found, should we just `await` this step on each redeem request rather than handling the error?

Reply:
Even better, `waitForOpreturn` returns the result of `getTxIdByOpReturn` so I think the `getTxIdByOpReturn` can be replaced by `waitForOpreturn`.

Reply:
> Even better, `waitForOpreturn` returns the result of `getTxIdByOpReturn` so I think the `getTxIdByOpReturn` can be replaced by `waitForOpreturn`.

Ah, I did not notice that. Thank you.

And yes, we do need to wait (sometimes) as the tests above just send the redeem request, but do not await for those to be broadcast / confirmed. So in this test, where we look at the output, we may have to wait.